js/interface.js

```diff
@@ -287,6 +273,23 @@ Fliplet.Widget.onSaveRequest(function() {
   save(true);
 });
 
+Fliplet.Widget.onCancelRequest(function() {
+  if (providerInstance) {
```

Comment:
if you have both `providerInstance` and `emailTemplateAddProvider` it looks like you're forwarding the event to both. You sure it needs to work like that?

Reply:
Any suggestion? @squallstar

Reply:
1. what is `providerInstance`
2. maybe check first for `emailTemplateAddProvider` and if exists, cancel that one, return (on this logic) and make it `null` or something.